src/test/java/co/rsk/federate/FedNodeRunnerTest.java

```diff
@@ -516,34 +515,26 @@ private SignerConfig getHSMBTCSignerConfig(int version) throws HSMClientExceptio
                     NetworkParameters.ID_UNITTESTNET.toString()));
         }
 
-        return btcSignerConfig;
+        return new SignerConfig("BTC", configBuilder.build());
     }
 
     private SignerConfig getHSMRSKSignerConfig() {
-        SignerConfig rskSignerConfig = mock(SignerConfig.class);
-        Config hsmConfig = mock(Config.class);
-        when(rskSignerConfig.getId()).thenReturn("RSK");
-        when(rskSignerConfig.getType()).thenReturn("hsm");
-        when(rskSignerConfig.getConfig()).thenReturn(hsmConfig);
-        when(hsmConfig.hasPath("host")).thenReturn(true);
-        when(hsmConfig.getString("host")).thenReturn("127.0.0.1");
-        when(hsmConfig.hasPath("port")).thenReturn(true);
-        when(hsmConfig.getInt("port")).thenReturn(9999);
-        when(hsmConfig.getString("keyId")).thenReturn("m/44'/137'/0'/0/0");
-        return rskSignerConfig;
+        Config config = TestConfigBuilder.builder()
+            .withValue("type", "hsm")
+            .withValue("host", "127.0.0.1")
+            .withValue("port", 9999)
+            .withValue("keyId", "m/44'/137'/0'/0/0")
+            .build();
+        return new SignerConfig("RSK", config); 
     }
 
     private SignerConfig getHSMMSTSignerConfig() {
-        Config hsmConfig = mock(Config.class);
-        SignerConfig mstSignerConfig = mock(SignerConfig.class);
-        when(mstSignerConfig.getId()).thenReturn("MST");
-        when(mstSignerConfig.getType()).thenReturn("hsm");
-        when(mstSignerConfig.getConfig()).thenReturn(hsmConfig);
-        when(hsmConfig.hasPath("host")).thenReturn(true);
-        when(hsmConfig.getString("host")).thenReturn("127.0.0.1");
-        when(hsmConfig.hasPath("port")).thenReturn(true);
-        when(hsmConfig.getInt("port")).thenReturn(9999);
-        when(hsmConfig.getString("keyId")).thenReturn("m/44'/137'/1'/0/0");
-        return mstSignerConfig;
+        Config config = TestConfigBuilder.builder()
+            .withValue("type", "hsm")
+            .withValue("host", "127.0.0.1")
+            .withValue("port", 9999)
+            .withValue("keyId", "m/44'/137'/1'/0/0")
```

Comment:
Perhaps all these could be part of a single method?

`withHsmSigner(host, port, keyid)` where the `type` is automatically set to `hsm`

We could also have one with default values `withHsmSigner()` where it uses all these values by default

Reply:
Sounds good, what I have done is added the method with `host` and `port` already set except the `keyId`. The keyId changes depending from `BTC, RSK, MST` so a default here will not be possible. Wdyt with how is set up currently?